app/scripts/lib/ens-ipfs/setup.js

```diff
@@ -40,12 +40,13 @@ function setupEnsIpfsResolver ({ provider }) {
   }
 
   async function attemptResolve ({ tabId, name, path, search, fragment }) {
+    const ipfsGateway = getIpfsGateway()
     extension.tabs.update(tabId, { url: `loading.html` })
     let url = `https://app.ens.domains/name/${name}`
     try {
       const { type, hash } = await resolveEnsToIpfsContentId({ provider, name })
       if (type === 'ipfs-ns') {
-        const resolvedUrl = `https://gateway.ipfs.io/ipfs/${hash}${path}${search || ''}${fragment || ''}`
+        const resolvedUrl = `https://${hash}.${ipfsGateway}${path}${search || ''}${fragment || ''}`
```

Comment:
Does this format work with the old gateway? That is, can someone still use `gateway.ipfs.io`?

Reply:
I don't think so. In my understanding the motivating issue encourages us to forbid using that gateway: #5724 

Perhaps we need to explicitly disallow it?

Reply:
> Perhaps we need to explicitly disallow it?

We could, but maybe that's a separate issue, if we're not trying to support it then this makes sense.

Reply:
That's what I did, fix incoming.

Reply:
(sorry for being late, this is probably something for a separate issue/PR now, feel free to copy/paste in correct place) 

To make this more bullet proof, you could add a quick validation, to ensure provided hostname is a real [subdomain gateway](https://docs.ipfs.io/guides/guides/addressing/#subdomain-gateway), and refuse to save invalid ones.

Try fetching a small, well-known hash (for example: string `hello` is `bafkreicysg23kiwv34eg2d7qweipxwosdo2py4ldv42nbauguluen5v6am`) and see if returned value matches expectations:

- https://bafkreicysg23kiwv34eg2d7qweipxwosdo2py4ldv42nbauguluen5v6am.ipfs.dweb.link
- https://bafkreicysg23kiwv34eg2d7qweipxwosdo2py4ldv42nbauguluen5v6am.ipfs.cf-ipfs.com

This would also protect you from users setting [path-based gateways](https://docs.ipfs.io/guides/guides/addressing/#path-gateway) (which do not provide Origin-based isolation.

Reply:
@lidel thanks for the suggestion, and no worries about missing the merge. New issue here: #7698